### PR TITLE
TiledSpriteObject is displayed correctly on Android devices

### PR DESCRIPTION
**Pull request by victorlevasseur**
(manual repetition using vertices instead of texture repetition)

### CODE REVIEW

File: Extensions/TiledSpriteObject/TiledSpriteObject.cpp [hunk 1]
```diff
@@ -107,8 +107,63 @@ bool RuntimeTiledSpriteObject::Draw( sf::RenderTarget& window )
     if ( hidden ) return true;
     if(!texture) return true;
 
+#if defined(ANDROID)
+    const unsigned int textureWidth = texture->texture.getSize().x;
+    const unsigned int textureHeight =  texture->texture.getSize().y;
+    std::vector<sf::Vertex> vertices;
+    vertices.resize(
```
Comment (4ian) [inline Extensions/TiledSpriteObject/TiledSpriteObject.cpp]: Nice! :+1:

**Comment by 4ian** [comment]
As there will be a lot of vertices, it's a good idea to resize the array before inserting vertices into it.
Also, you can reduce the number of vertices to draw for each repetition : use sf::TrianglesStrip. It works like a strip/ribbon of triangles (each new vertice will create a triangle using the last two vertices).
So you can draw a quad by drawing the vertices in this order : top left, top right, bottom left, bottom right.

I guess that it's important to apply here this optimisation as lots of vertices will be drawn.

**Comment by 4ian** [comment]
You could even draw a whole row of texture by drawing :
(Texture first repetition:) top left, bottom left, top right, bottom right then (Texture 2nd repetition:) top right, bottom right then (Texture 3rd repetition:) top right, bottom right ...

**Comment by victorlevasseur** [comment]
I'll try to do that. Meanwhile, I think we can merge it (it's working at least, and isn't that bad).

**Comment by 4ian** [comment]
Yep I merged this as it's working anyway :+1: In fact, the way of drawing I was talking about will surely require more calls to Draw so in the end I'm not sure it's necessarily better.

**Pull request merged**